docs/tutorials/essentials/part-7-rtk-query-basics.md

```diff
@@ -0,0 +1,692 @@
+
+Currently, `<PostsList>` is specifically importing `useSelector`, `useDispatch`, and `useEffect`, reading posts data and loading state from the store, and dispatching the `fetchPosts()` thunk on mount to trigger the data fetch. **The `useGetPostsQueryHook` replaces all of that!**
+
+Let's see how `<PostsList>` looks when we use this hook:
+
+```jsx title="features/posts/PostsList.js"
+import React from 'react'
+import { Link } from 'react-router-dom'
+
+import { Spinner } from '../../components/Spinner'
+import { PostAuthor } from './PostAuthor'
+import { TimeAgo } from './TimeAgo'
+import { ReactionButtons } from './ReactionButtons'
+
+// highlight-next-line
+import { useGetPostsQuery } from '../api/apiSlice'
+
+// highlight-next-line
+let PostExcerpt = ({ post }) => {
+  return (
+    <article className="post-excerpt" key={post.id}>
+      <h3>{post.title}</h3>
+      <div>
+        <PostAuthor userId={post.user} />
+        <TimeAgo timestamp={post.date} />
+      </div>
+      <p className="post-content">{post.content.substring(0, 100)}</p>
+
+      <ReactionButtons post={post} />
+      <Link to={`/posts/${post.id}`} className="button muted-button">
+        View Post
+      </Link>
+    </article>
+  )
+}
+
+export const PostsList = () => {
+  // highlight-start
+  const {
+    data: posts,
+    isLoading,
+    isSuccess,
+    isError,
+    error
+  } = useGetPostsQuery()
+  // highlight-end
+
+  let content
+
+  // highlight-next-line
+  if (isLoading) {
+    content = <Spinner text="Loading..." />
+    // highlight-next-line
+  } else if (isSuccess) {
+    content = posts.map(post => <PostExcerpt key={post.id} post={post} />)
+    // highlight-next-line
+  } else if (isError) {
+    content = <div>{error.toString()}</div>
+  }
+
+  return (
+    <section className="posts-list">
+      <h2>Posts</h2>
+      {content}
+    </section>
+  )
+}
+```
+
+Conceptually, `<PostsList>` is still doing all the same work it was before, but we were able to replace the multiple `useSelector` calls and the `useEffect` dispatch with a single call to `useGetPostsQuery()`.
+
+Each generated query hook returns an object containing several fields, including:
+
+- `data`: the actual response contents from the server. **This field will be `undefined` until the response is received**.
+- `isLoading`: a boolean indicating if the hook is currently making the _first_ request to the server for this data
+- `isFetching`: a boolean indicating if the hook is currently making _any_ request to the server
+- `isSuccess`: a boolean indicating if the hook has made a successful request and has cached data available (ie, `data` should be defined now)
+- `isError`: a boolean indicating if the last request had an error
+- `error`: a serialized error object
+
+Typically, you'll want to destructure fields from the result object, and possibly rename `data` to a more specific variable like `posts` to describe what it contains.
```

Comment:
Not 10000% sure we should encourage destructuring *too* much as it has drawbacks using TS (if destructured, `isSuccess` will no longer indicate that `data` is not `undefined`)

Reply:
Will this be fixed in TS 4.4?

I find destructuring to be very natural here, and I would find it annoying to always have to do `res.isSuccess` all the time.

Reply:
> Will this be fixed in TS 4.4?

Not to my knowledge, 4.4 doesn't change anything about that. It still needs a single object to maintain a relationship between properties

mini-example on 4.4.2 here: https://tsplay.dev/NBP1xW